### DISCUSSION TITLE
Plume/contrail overlap analysis in high density airspace

### DISCUSSION THREAD

**Discussion by ktait98** [contrailcirrus/pycontrails#63] [Ideas]
Since the Pycontrails workshop, I have been mulling over a number of different ways in which we can better accommodate for multi-aircraft cumulative effects, that affect climate impact in high density regions (e.g. NOx saturation and contrail dehydration effects). 

One way I envisage this is by modelling individual aircraft plumes as 4d meshes (3d in space, updated every timestep), which are then 'voxelised' to convert them to an axis-aligned common 4D grid space. Each of the occupied grid cells will have properties corresponding to the contrail properties at the midpoint of the cell. This would then allow for analysis of the overlap effects of arbitrarily oriented/sized aircraft plumes in 4D space, as overlapping plume segments will input properties into common grid cells. Beyond this, simulations can be carried out to determine the chemical/physical/dynamical evolution of the contents of occupied grid cells.

I'm wondering if anyone in the Pycontrails community would like to share their experience of carrying out similar studies (e.g. in the covid study by Schumann et al. (2021)), or whether there are already mechanisms in place which allow for such computations. I am aware of the cocipgrid directory, which appear to take some steps towards implementing this already. 

If anyone has any further documentation on how cocipgrid is implemented, and/or has any thoughts pertaining to the idea of modelling overlap of arbitrary aircraft plumes using Pycontrails, please get involved!

**Comment by zebengberg** [comment]
Thanks for starting this discussion!

@roger-teoh and I are hoping to undertake a detailed study looking into contrail-contrail overlap effects later this year. It'd be great to get help with this. I'm not sure exactly what overlap effects we'd focus on (certainly radiative forcing, hopefully something pertaining to dehydration effects).

I think you've roughly described the approach we'd like to take. We'd essentially run a CoCiP simulation on a fleet over a high traffic region (for example, Europe). At each time step of the contrail evolution, we'd construct a fine grid (much finer than ECMWF weather data) and insert contrail data according to the precise CoCiP geometry (a Gaussian plume with a width, depth, and direction of travel at each waypoint). This gridded data would then inform the next step in the CoCiP evolution. Roger has already run a coarser version of this in which he doesn't use exact contrail geometry (he is currently integrating some of this work back into pycontrails). We would add new code to support this fine-grained simulation as well.

**Reply by ktait98** [reply]
Hi @zebengberg @roger-teoh

Thanks for getting back to me so quickly. I was planning to carry out a study similar to this, for my final PhD study. The aim is to utilise Pycontrails along side our in-house chemical scheme at Bristol, to investigate plume overlap effects. Based on existing literature, I have identified the three primary saturation effects to be:
- contrail-contrail radiative effects due to overlapping contrails at different flight levels,
- contrail-contrail dehydration effects due to mutual inhibition of available water vapour,
- NOx saturation leading to reductions in ozone production efficiency.

I have a relatively well-defined approach to conducting this study, but I don't want to repeat what is already in the works. Perhaps there is an opportunity here to share ideas and potentially collaborate on such a study? I have until the end of this year to complete my research before thesis writing commences, so I will be working on this full time until then. 

If either/both of you have the time to meet in the coming weeks to discuss this further, please do let me know by email (kieran.tait@bristol.ac.uk).
